content/docs/replication/migrating-volumes.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Migrating Volumes
+linktitle: Migrating Volumes
+weight: 6
+description: >
+  Migrating Volumes Between Storage Classes
+---
+
+You can migrate existing, already provisioned, volumes to another storage class by using volume migration feature. 
+
+As of CSM 1.3 two versions of migration supported: 
+- To replicated storage class from NON replicated one
+- To NON replicated storage class from replicated one
+
+## Prerequisites
+- Original volume is from the one of currently supported CSI drivers (see Support Matrix)
+- Migrated sidecar is installed alongside with the driver, you can enable it in your `myvalues.yaml` file
+```yaml
+migration:
+  enabled: true
+```
+
+## Support Matrix
+| Migration Type | PowerMax | PowerStore | PowerScale | PowerFlex | Unity | 
+| - | - | - | - | - | - | 
+| NON_REPL_TO_REPL | Yes | No | No | No | No |
+| REPL_TO_NON_REPL | Yes | No | No | No | No |
+
+
+## Basic Usage
+
+To trigger migration procedure, you need to patch existing PersistentVolume with migration annotation (by default `migration.storage.dell.com/migrate-to`) and in value of said annotation specify StorageClass name you want to migrate to. 
+
+For example, if have PV named `test-pv` already provisioned and we want to migrate it to replicated storage class named `powermax-replication` we can run:
+
+```shell
+kubectl patch pv test-pv -p '{"metadata": {"annotations":{"migration.storage.dell.com/migrate-to":"powermax-replication"}}}'
+```
+
+Patching PV resource will trigger migration sidecar that will call `VolumeMigrate` call from the CSI driver. After migration is finished new PersistentVolume will be created in cluster with name of original PV plus `-to-<sc-name>` appended to it. 
+
+So, in case of our example, we will see the following when running `kubectl get pv`: 
+```shell
+NAME                                   CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                       STORAGECLASS                REASON   AGE
+test-pv                                1Gi        RWO            Retain           Bound       default/test-pvc            powermax                             5m
+test-pv-to-powermax-replication        1Gi        RWO            Retain           Available                               powermax-replication                 10s
+
+```
+
+When Volume Migration is finished source PV will be updated with EVENT that denotes that. 
+
+Newly created PV (`test-pv-to-powermax-replication` in our example) is available for consumption via static provisioning by any PVC that will request it.
```

Comment:
Post migration, will we support other use-cases like volume expansion, snapshot creation?